### PR TITLE
Grid editor span operations

### LINKED ISSUE #101: Implement split span
Implement split spans.

Will include identification of different parts of the same span via, e.g., some sort of visual feedback.

### LINKED ISSUE #102: Implement merge span
Implement merging 2 or more spans into one.

Annotations will have to be merged, conflicting annotations caught and fed back to the user.

### PR DESCRIPTION
**Pull request by sdruskat**
## Pull request type

 

- Feature

## What is the current behavior?

Issue Number: #101, #102

## What is the new behavior?

- Fixes #101 
- Fixes #102
- Existing spans can be split, i.e., for each token covered by the existing spans, a new span is created with the existing span annotations, and the original span removed.
- Existing spans can be merged, i.e. > 1 span within the same annotation column with the same value for the annotation name of the column can be merged into one span when selected.

## Does this introduce a breaking change?

- No

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added **or** the PR is neither a bug fix nor a feature
- [x] Docs have been reviewed and added / updated if needed **or** the PR is neither a bug fix nor a feature
- [x] The *Unreleased* section in CHANGELOG.md has been amended to reflect the changes in this PR if needed
- [x] Build (`mvn verify`) was run locally and any changes were pushed
- [x] The pull request is against the correct branch (`master` for bug fixes, `develop` for new functionality)
- [x] [Dependencies and citation templates](https://github.com/hexatomic/hexatomic/tree/develop/releng/templates) have been updated where necessary **or** there are no new dependencies

---

# Checklist for maintainers

## Releases

- Do **NOT** push the green merge button on GitHub.  
Follow the latest version of the [developer/maintainer docs](https://hexatomic.github.io/hexatomic/dev/) for making releases.
- Check that license and citation information for dependencies are complete (each dependency has a folder in `/THIRD-PARTY/`).

### CODE REVIEW

File: bundles/org.corpus_tools.hexatomic.grid/src/main/java/org/corpus_tools/hexatomic/grid/GridHelper.java [hunk 1]
```diff
@@ -99,26 +101,141 @@ public static boolean areSelectedCellsInSingleSpanColumn(
...
+   * @param annoName The name string
+   * @return The valid qualified annotation name
+   */
+  public static String getAnnotationQName(String annoNamespace, String annoName) {
+    if (annoNamespace == null || annoNamespace.isEmpty()) {
+      return annoName;
+    } else {
+      return annoNamespace + SaltUtil.NAMESPACE_SEPERATOR + annoName;
+    }
+  }
```
Comment (thomaskrause) [inline bundles/org.corpus_tools.hexatomic.grid/src/main/java/org/corpus_tools/hexatomic/grid/GridHelper.java:176]: This does not handle the case that `annoName` is `null`. With this additional constraint, the whole function would be exactly the same as `SaltUtil#createQName`. Is there a specific reason to copy the implementation or can we just use the Salt function?

```suggestion
```

File: bundles/org.corpus_tools.hexatomic.grid/src/main/java/org/corpus_tools/hexatomic/grid/internal/data/GraphDataProvider.java [hunk 2]
```diff
@@ -681,18 +691,95 @@ public void createEmptyAnnotationSpan(Set<PositionCoordinate> selectedCoordinate
...
+   * 
+   * @param span The {@link SSpan} to split
+   * @param coordinates The coordinates for the {@link SSpan} to split
+   */
+  public void splitAnnotationSpan(SSpan span, PositionCoordinate[] coordinates) {
+    PositionCoordinate firstCoord = coordinates[0];
+    int firstCol = firstCoord.getColumnPosition();
+    String annoNamespace = getAnnotationNamespace(firstCol);
+    String annoName = getAnnotationName(firstCol);
+    String annoQName = GridHelper.getAnnotationQName(annoNamespace, annoName);
```
Comment (thomaskrause) [inline bundles/org.corpus_tools.hexatomic.grid/src/main/java/org/corpus_tools/hexatomic/grid/internal/data/GraphDataProvider.java:709]: If the custom helper function is removed, we could use the SaltUtil helper function here.

```suggestion
    String annoQName = SaltUtil.createQName(annoNamespace, annoName);
```

**Review by thomaskrause — APPROVED** [review]
Looks good to me, but you might not want to re-implement `SaltUtils.createQName`

File: bundles/org.corpus_tools.hexatomic.grid/src/main/java/org/corpus_tools/hexatomic/grid/GridHelper.java (hunk 1 above)
Comment (sdruskat) [reply bundles/org.corpus_tools.hexatomic.grid/src/main/java/org/corpus_tools/hexatomic/grid/GridHelper.java:176]: Thanks! I'll update to use the Salt function indeed. With the number of helper classes strewn across plugins and projects, I couldn't find this specific one ;). The "create" prefix probably flung me off the trail ...

**Pull request merged**